migen/build/platforms/tinyfpga_a.py

```diff
@@ -0,0 +1,99 @@
+import bisect
+
+from migen import *
+from migen.build.generic_platform import *
+from migen.build.lattice import LatticePlatform
+
+_io = [
+]
+
+_connectors = [
+    # Pins 1-22
+    ("GPIO", "13 14 16 17 20 21 23 25 26 27 28 4 5 8 9 10 11 12"),
+]
+
+
+# Default peripherals.
+# Only on AX.
+led = [
+    ("user_led", 0, Pins("GPIO:0"))
+]
+
+serial = [
+    ("serial", 0,
+        Subsignal("tx", Pins("GPIO:2")),
+        Subsignal("rx", Pins("GPIO:3")),
+        IOStandard("LVCMOS33")
+     )
+]
+
+
+class MachClock(Module):
+    # Diamond will complain that the frequency of the oscillator doesn't match
+    # if it doesn't match one of the below frequencies. Since we tend to deal
+    # with periods in nanoseconds, choose the closest frequency (in MHz)
+    # without going over. These numbers were extracted from
+    # "MachXO2 sysCLOCK PLL Design and Usage Guide"
+    supported_freqs = [
+        2.08, 2.15, 2.22, 2.29, 2.38, 2.46, 2.56, 2.66, 2.77, 2.89,
+        3.02, 3.17, 3.33, 3.50, 3.69, 3.91, 4.16, 4.29, 4.43, 4.59,
+        4.75, 4.93, 5.12, 5.32, 5.54, 5.78, 6.05, 6.33, 6.65, 7.00,
+        7.39, 7.82, 8.31, 8.58, 8.87, 9.17, 9.50, 9.85, 10.23, 10.64,
+        11.08, 11.57, 12.09, 12.67, 13.30, 14.00, 14.78, 15.65, 15.65, 16.63,
+        17.73, 19.00, 20.46, 22.17, 24.18, 26.60, 29.56, 33.25, 38.00, 44.33,
+        53.20, 66.50, 88.67, 133.00
+    ]
+
```

Comment:
You should maybe try to keep the same presentation than in the datasheet:

![image](https://user-images.githubusercontent.com/1450143/40296471-f6fce6ac-5cdc-11e8-96a5-686ee294b156.png)

-->

````
supported_freqs = [
    2.08, 4.16,  8.31,  15.65,
    2.15, 4.29,  8.58,  16.63,
    2.22, 4.43,  8.87,  17.73,
    2.29, 4.59,  9.17,  19.00,
    2.38, 4.75,  9.50,  20.46,
    2.46, 4.93,  9.85,  22.17,
    2.56, 5.12, 10.23,  24.18,
    2.66, 5.32, 10.64,  26.60,
    2.77, 5.54, 11.08,  29.56,
    2.89, 5.78, 11.57,  33.25,
    3.02, 6.05, 12.09,  38.00,
    3.17, 6.33, 12.67,  44.33,
    3.33, 6.65, 13.30,  53.20,
    3.50, 7.00, 14.00,  66.50,
    3.69, 7.39, 14.78,  88.67,
    3.91, 7.82, 15.65, 133.00
]
````